From d93e2cda6f58b321364bd951d5de1f1d663ea3c1 Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Sat, 9 Oct 2021 12:24:44 -0700
Subject: [PATCH 81/85] tools/meson: fix usage with SDK

The python path cannot be embedded in the meson binary as it changes
with the SDK.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
---
 include/meson.mk     | 2 +-
 tools/meson/Makefile | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/meson.mk b/include/meson.mk
index 9862f0e662..12e38b9e71 100644
--- a/include/meson.mk
+++ b/include/meson.mk
@@ -56,7 +56,7 @@ MESON_CPU:="$(CPU_TYPE)$(if $(CPU_SUBTYPE),+$(CPU_SUBTYPE))"
 endif
 
 define Meson
-	$(2) $(STAGING_DIR_HOST)/bin/meson $(1)
+	$(2) $(STAGING_DIR_HOST)/bin/$(PYTHON) $(STAGING_DIR_HOST)/bin/meson $(1)
 endef
 
 define Meson/CreateNativeFile
diff --git a/tools/meson/Makefile b/tools/meson/Makefile
index 0cf0efba47..1aab5d8fcb 100644
--- a/tools/meson/Makefile
+++ b/tools/meson/Makefile
@@ -21,7 +21,7 @@ endef
 
 define Host/Install
 	$(INSTALL_DIR) $(STAGING_DIR_HOST)/bin
-	$(HOST_BUILD_DIR)/packaging/create_zipapp.py $(HOST_BUILD_DIR) --interpreter $(STAGING_DIR_HOST)/bin/$(PYTHON) --outfile $(STAGING_DIR_HOST)/bin/meson
+	$(HOST_BUILD_DIR)/packaging/create_zipapp.py $(HOST_BUILD_DIR) --outfile $(STAGING_DIR_HOST)/bin/meson
 	$(INSTALL_DIR) $(STAGING_DIR_HOST)/lib/meson
 	$(INSTALL_CONF) files/openwrt-cross.txt.in $(STAGING_DIR_HOST)/lib/meson/
 	$(INSTALL_CONF) files/openwrt-native.txt.in $(STAGING_DIR_HOST)/lib/meson/
-- 
2.35.1

